script.js

```diff
@@ -1,4 +1,299 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+var cardName = [
+  "ace",
+  "two",
+  "three",
+  "four",
+  "five",
+  "six",
+  "seven",
+  "eight",
+  "nine",
+  "ten",
+  "jack",
+  "queen",
+  "king",
+];
+
+var suits = ["Hearts", "Diamonds", "Clubs", "Spades"];
+
+// To create an array to store the cards drawn by player >> playerHand [i]
+// To create a function that totals up the value of the cards in the array
+// To win, playerTotal == 21 or more than dealer but still below 21
+
+// Game Mode
+var mode = "hit";
+function hit() {
+  playerDraws();
+  computerDraws();
+  return checkPlayerCards();
+}
+
+// Player and computer hand as an array to track cards drawn
+var playerHand = [];
+var computerHand = [];
+
+function playerDraws() {
+  var card = shuffledDeck.pop();
+  playerHand.push(card);
+}
+
+function aceOneOrEleven() {
+  if (handTotal(playerHand) <= 11) {
+    for (var i = 0; i < playerHand.length; i += 1) {
+      if (playerHand[i].rank == 1) {
+        playerHand[i].rank = 11;
+        i = playerHand.length;
+      }
+    }
+    return checkPlayerCards();
+  } else {
+    return checkPlayerCards();
+  }
+}
+
+// Variable to state the hand total of both player and computer
+function compareHand() {
+  var compareHandStatement = `Player's card total is ${handTotal(
+    playerHand
+  )}. Computer's card total is ${handTotal(computerHand)}.`;
+  return compareHandStatement;
+}
+
+// Computer will draw as long as it's below 21
+function computerDraws() {
+  var computerRiskTolerance = 21 - getRandomIndex(3);
+  if (handTotal(computerHand) < computerRiskTolerance) {
+    var card = shuffledDeck.pop();
+    computerHand.push(card);
+  }
+}
+
+// Calculator function to calculate number player has
+function handTotal(array) {
+  var sum = 0;
+  for (var i = 0; i < array.length; i += 1) {
+    sum += array[i].rank;
+  }
+  return sum;
+}
+
+// Win by getting 21 condition
+function winCondition(array) {
+  if (handTotal(array) == 21) {
```

Comment:
You almost never need to return true/ false. You can just return the expression itself e.g. 
return handTotal(array) == 21